src/sparse/impl/KokkosSparse_spmv_impl.hpp

```diff
@@ -271,6 +271,7 @@ int64_t spmv_launch_parameters(int64_t numRows, int64_t nnz, int64_t rows_per_th
     while(vector_length < max_vector_length && vector_length * 6 < nnz_per_row)
       vector_length*=2;
   }
+  vector_length=8;
```

Comment:
Is ``vector_length = 8`` meant for all backends? If it is, we can just take out the above logic about max_vector_length and the loop to choose it based on avg nnz per row.

Reply:
No, it was just a value that gave us reasonable results on matrices for the Kokkos Ecosystem paper but it would be better to test this out on a larger set of matrices and get a sense for a heuristic that works well on AMD.
I am not at all sure that 8 is good on other architectures... that's why this is definitely just a draft that I need to work on x)